src/audio/module_adapter/module/waves.c

```diff
@@ -356,8 +356,10 @@ static int waves_effect_init(struct processing_module *mod)
 	waves_codec->o_format = waves_codec->i_format;
 
 	waves_codec->sample_size_in_bytes = sample_bytes;
-	waves_codec->buffer_bytes = mod->period_bytes;
-	waves_codec->buffer_samples = waves_codec->buffer_bytes / src_fmt->channels /
+	// Prepare a buffer for 1 period worth of data
+	// dev->pipeline->period is in us
+	waves_codec->buffer_samples = src_fmt->rate * dev->pipeline->period / 1000000;
```

Comment:
Please use C style comments ```/* ... *```. Also, please fix typo in commit subject.

Reply:
Updated.